File: F/solution/test/TLE.py
import sys

R, C, Q = map(int, sys.stdin.readline().split())

image = [[0 for cols in range(C + 1)] for rows in range(R + 1)]
for r in range(1, R + 1):
    cols = list(map(int, sys.stdin.readline().split()))
    for c in range(1, C + 1):
        image[r][c] = cols[c - 1]

for q in range(Q):
    r1, r2, c1, c2 = map(int, sys.stdin.readline().split())

    n = (r2 - r1 + 1) * (c2 - c1 + 1)
    sum = 0
    for r in range(r1, r2 + 1):
        for c in range(c1, c2 + 1):
            sum += image[r][c]

    print(sum // n)

File: F/solution/test/TLE.java
import java.util.*;
import java.io.*;

public class Solution {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		StringTokenizer st = new StringTokenizer(br.readLine());
		int R = Integer.parseInt(st.nextToken());
		int C = Integer.parseInt(st.nextToken());
		int Q = Integer.parseInt(st.nextToken());

		int[][] image = new int[R + 1][C + 1];
		for (int r = 1; r <= R; r++) {
			st = new StringTokenizer(br.readLine());
			for (int c = 1; c <= C; c++)
				image[r][c] = Integer.parseInt(st.nextToken());
		}

		while (Q-- > 0) {
			st = new StringTokenizer(br.readLine());
			int r1 = Integer.parseInt(st.nextToken());
			int c1 = Integer.parseInt(st.nextToken());
			int r2 = Integer.parseInt(st.nextToken());
			int c2 = Integer.parseInt(st.nextToken());

			int sum = 0;
			for (int r = r1; r <= r2; r++)
				for (int c = c1; c <= c2; c++)
					sum += image[r][c];

			int n = (r2 - r1 + 1) * (c2 - c1 + 1);
			bw.write(String.valueOf(sum / n));
			bw.write('\n');
		}

		bw.close();
	}
}
